src/Estimators/include/BipedalLocomotionControllers/Estimators/RecursiveLeastSquare.h

```diff
@@ -0,0 +1,118 @@
+/**
+ * @file RecursiveLeastSquare.h
+ * @authors Giulio Romualdi
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_CONTROLLERS_ESTIMATORS_RLS_H
+#define BIPEDAL_LOCOMOTION_CONTROLLERS_ESTIMATORS_RLS_H
+
+#include <functional>
+#include <memory>
+
+#include <iDynTree/Core/MatrixDynSize.h>
+#include <iDynTree/Core/VectorDynSize.h>
+
+#include <BipedalLocomotionControllers/ParametersHandler/IParametersHandler.h>
+
+namespace BipedalLocomotionControllers
+{
+namespace Estimators
+{
+/**
+ * RecursiveLeastSquare contains the implementation of the Recursive least square filter described
+ * in Lennart Ljung - System Identification Theory for the User (1999, Prentice Hall). Chapter 11
+ * Section 2.
+ */
+class RecursiveLeastSquare
+{
+    iDynTree::VectorDynSize m_state; /**< Vector containing the expected value of the estimated
+                                        state */
+    iDynTree::VectorDynSize m_measuraments; /**< Vector containing the measurements */
+
+    iDynTree::MatrixDynSize m_stateCovarianceMatrix; /**< Covariance matrix of the state */
+
+    /** Covariance matrix of the measuraments we assume that the measuraments are uncorrelated
+     (furthermore since the model of the noise is Gaussian the random variable are also independent)
+     Since the variable are independent we are interested only on the element in the diagonal
+     of the matrix */
+    iDynTree::MatrixDynSize m_measurementCovarianceMatrix;
+
+    iDynTree::MatrixDynSize m_kalmanGain; /**< Gain of the Kalman filter */
+
+    double m_lambda; /**< Filter gain. The recursive least square filter is equivalent to a kalman
+                        filter if lambda is equal to 1 */
+
+    std::function<iDynTree::MatrixDynSize(void)> m_regressor; /**< Function containing the regressor
+                                                                 of the system */
+
+    /**
+     * Enumerator useful to described the current status of the filter
+     */
+    enum class State
+    {
+        NotInitialized, /**< The filter is not initialized yet please class
```

Comment:
"please use"

Reply:
Done in 260a0a8ccc47816aa894609fa3acfac422eb6b41